nixos/modules/programs/gnupg.nix

```diff
@@ -0,0 +1,61 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.gnupg;
+
+in
+
+{
+
+  options.programs.gnupg = {
+    agent.enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Start GnuPG agent automatically for every user session.
+      '';
+    };
+
+    agent.enableSSHSupport = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable SSH agent support in GnuPG agent. Also sets SSH_AUTH_SOCK
+        environment variable correctly.
+      '';
+    };
+  };
+
+  config = mkIf cfg.agent.enable {
+    systemd.user.services.gpg-agent = {
+      description = "GnuPG Agent";
+      wantedBy = [ "default.target" "graphical.target" ];
+      serviceConfig = {
+        Type = "forking";
+        ExecStart = "${pkgs.gnupg}/bin/gpg-agent --daemon "
+          + optionalString cfg.agent.enableSSHSupport "--enable-ssh-support";
+        PassEnvironment = "DISPLAY";
```

Comment:
otherwise passing `--systemd` to dbus-update-activation-environment might be a good idea: https://dbus.freedesktop.org/doc/dbus-update-activation-environment.1.html